internal/preload/azure_types.go

```diff
@@ -0,0 +1,15 @@
+package preload
+
+var AzureInstanceType instanceType
+
+func init() {
+	AzureInstanceType = instanceType{
+		filename: "azure_types.yaml",
+		path:     "azure_availability",
+		etagName: "azure-types",
+	}
+	err := AzureInstanceType.Load()
+	if err != nil {
+		panic(err)
```

Comment:
You could look up in the `Load` function, it shows: `fmt.Errorf("unable to read instance types %s: %w", p.filename, err)`

I believe this is sufficient, it contains the error message, filename and root error (e.g. "not found").

Note I use `panic` because this is an `init` function. The app hasn’t been initialized yet, there is no logging, no configuration. Therefore I use panic to simply exit immediately. Since these YAML files are embedded into the application itself, this will always work (or always fail if a file was renamed or something).

If you still want more detailed error message, let me know, what exactly should I put here. Panic also shows stacktrace, should be pretty obvious where it failed.

Reply:
How about this: `panic(fmt.Errorf("cannot preload ec2 types: %w", err))`